lib/IMAP/PreviewEnhancer.php

```diff
@@ -59,6 +66,22 @@ public function process(Account $account, Mailbox $mailbox, array $messages): ar
 			return array_merge($carry, [$message->getUid()]);
 		}, []);
 
+		if ($preLoadAvatars) {
+			foreach ($messages as $message) {
+				$from = $message->getFrom()->first();
+				if ($message->getAvatar() === null && $from !== null && $from->getEmail() !== null && $userId !== null) {
```

Comment:
nit: negate the condition and use `continue` to skip one loop iteration. That avoids at least one level of indentation in this deep indentation tree.